modules/data.remote/inst/s2lai.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Calculates LAI from Sentinel 2 data. Uses geeapi to download data and the python implementation of the SNAP toolbox to calculate LAI.
+
+Requires Python3
+
+Warning: Point coordinates as input has currently not been implemented.
+
+Author: Ayush Prasad
+"""
+
+import satellitetools.biophys_xarray as bio
+from satellitetools import gee
+import geopandas as gpd
+import os
+
+
+def s2_lai(geofile, outdir, start, end, qi_threshold):
+    """ 
+    Downloads sentinel 2 data, calculates LAI and saves it in a netCDF file at the specified location.
+    
+    Parameters
+    ----------
+    
+    geofile (str) -- path to the file containing the coordinates and name of the AOI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date of the data request in the form YYYY-MM-DD
+    
+    qi_threshold (float) -- From satellitetools: Threshold value to filter images based on used qi filter. qi filter holds labels of classes whose percentages within the AOI is summed. If the sum is larger then the qi_threshold, data will not be retrieved for that date/image. The default is 1, meaning all data is retrieved
+        
+ 
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory.
+            
+    Python dependencies required: earthengine-api, geopandas, pandas, netCDF4, xarray
+    
+    To test this function please run the following code inside a python shell after importing this module, testfile is included.
+    
+    s2_lai(geofile="./satellitetools/test.geojson", outdir="./out/", start="2019-01-01", end="2019-12-31", qi_threshold=1)       
```

Comment:
Ayush, I still see these PRs as an exercise where you play with Olli's functions and get a feeling of the pipeline, but just wanted to point it out early on that if/when we will generalize the module, we will need to think a bit more about the function names. I will post something on our gee-slack channel to explain what I mean.

For now, I'm fine with having this function in so that anyone who has a geojson file can already use it for getting LAI values for their ROI :+1: